### PR TITLE
Ignore NoNodeExists exceptions

### PR DESCRIPTION
**Pull request by jasonjckn**

### CODE REVIEW

File: src/jvm/storm/trident/topology/state/RotatingTransactionalState.java [hunk 1]
```diff
@@ -96,7 +98,16 @@ public void cleanupBefore(long txid) {
         SortedMap<Long, Object> toDelete = _curr.headMap(txid);
         for(long tx: new HashSet<Long>(toDelete.keySet())) {
             _curr.remove(tx);
-            _state.delete(txPath(tx));
+            try {
+                _state.delete(txPath(tx));
+            } catch(RuntimeException e) {
+                // Ignore NoNodeExists exceptions because when sync() it may populate _curr with stale data since
+                // zookeeper reads are eventually consistent.
+                if(!(e.getCause() instanceof KeeperException.NoNodeException)) {
```
Comment (nathanmarz) [inline src/jvm/storm/trident/topology/state/RotatingTransactionalState.java]: Let's replace this with logic similar to the exception-cause? function used in Clojure. That will be more robust than just checking one level deep in the exception stack.

**Comment by nathanmarz** [comment]
Just had a single comment about improving the code.

**Comment by jasonjckn** [comment]
Do you ever call clojure from java? Now we have code duplication with the above commit. 

If you'd like I can have clojure invoke the above java function.

**Comment by nathanmarz** [comment]
Yea, we should just have the clojure code call the Java version.

**Comment by nathanmarz** [comment]
Oh, I just realized this was against 0.9.0 branch. Make pull requests against master, not 0.9.0.

**Comment by jasonjckn** [comment]
ok, next time, please cherry-pick it.

**Pull request merged**